packages/frontend/app/services/channels/-utils/channel-factory.ts

```diff
@@ -23,11 +23,14 @@ export function buildChannelContextChain(
     id: contextChain.id,
     admin: buildChannelMember(contextChain.admin),
     members: contextChain.members.map((member) => buildChannelMember(member)),
-    supportingVote: buildVoteChain(contextChain.supportingVote),
+    supportingVote:
+      contextChain.previousChain === undefined && contextChain.supportingVote === undefined
+        ? undefined
+        : buildVoteChain(contextChain.supportingVote!),
```

Comment:
I think typescript is correct here, actually.
if previousChain is truthy, than buildVoteChain could be passed undefined...

Reply:
I need to figure this one out. The issue is that the only valid cases are when both previousChain and supportingVote don't exist or when both do exist. Having one exist and the other not is invalid.